### PR TITLE
One way TLS Supports Global SslParameters

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Support globally configured SslParameters when processing upstreams that use one-way TLS

# Context

In https://github.com/solo-io/gloo/pull/4283, we enabled users to define some upstream defaults for SslParameters at a global level, and allow those to be overwritten per upstream. 
This worked for upstreams that have an SslConfig: https://github.com/solo-io/gloo/blob/master/projects/gloo/pkg/translator/clusters.go#L108, however, there are some upstreams that implicitly use tls: https://github.com/solo-io/gloo/blob/master/projects/gloo/pkg/plugins/static/plugin.go#L120. This PR addresses that second case.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4285

### CODE REVIEW

**Review by saiskee — COMMENTED** [review]
Chatted with Sam about the ticket, looks good to me! @kdorosh to final review.

File: projects/gloo/pkg/plugins/static/plugin.go [hunk 1]
```diff
@@ -123,6 +126,7 @@ func (p *plugin) ProcessUpstream(params plugins.Params, in *v1.Upstream, out *en
 		// TODO: support client certificates
 		if out.TransportSocket == nil {
 			tlsContext := &envoyauth.UpstreamTlsContext{
+				CommonTlsContext: getCommonTlsContextFromUpstreamOptions(p.settings.GetUpstreamOptions()),
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/static/plugin.go:129]: don't we need to do this for aws and azure upstreams as well?

File: projects/gloo/pkg/plugins/static/plugin.go [hunk 2]
```diff
@@ -239,3 +247,17 @@ func metadataMatch(spec *v1static.UpstreamSpec, in *v1static.Host) *pbgostruct.S
 		},
 	}
 }
+
+// We support global UpstreamOptions to define SslParameters for all upstreams
+// If an upstream is configure with ssl, it will inherit the defaults here:
+// https://github.com/solo-io/gloo/blob/15da82bdd65ab4bcedbc7fb803ea0bb5f7e926fc/projects/gloo/pkg/translator/clusters.go#L108
+// However, if an upstream is configured with one-way TLS, we must explicitly apply the defaults, since there is no ssl
+// configuration on the upstream
+func getCommonTlsContextFromUpstreamOptions(options *v1.UpstreamOptions) (*envoyauth.CommonTlsContext, error) {
```
Comment (saiskee) [inline projects/gloo/pkg/plugins/static/plugin.go:256]: This is just a nit and is fine the way it is, but could you maybe DRY this by defining this function `plugin/utils/upstreamtls`?

File: projects/gloo/pkg/plugins/azure/plugin_test.go [hunk 3]
```diff
@@ -19,92 +19,195 @@ import (
...
+		out          *envoy_config_cluster_v3.Cluster
 	)
 
 	BeforeEach(func() {
 		var b bool
 		p = azureplugin.NewPlugin(&b)
-		p.Init(plugins.InitParams{Ctx: context.TODO()})
-		out = &envoy_config_cluster_v3.Cluster{}
+
+		// TODO: support namespace
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/azure/plugin_test.go:35]: remove todo?

**Review by kdorosh — APPROVED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**